src/Catalog/Helpers/GalleryDatabaseHelper.cs

```diff
@@ -0,0 +1,116 @@
+        }
+
+        /// <summary>
+        /// Returns a <see cref="SortedList{DateTime, IList{FeedPackageDetails}}"/> from the gallery database.
+        /// </summary>
+        /// <param name="keyDateFunc">The <see cref="DateTime"/> field to sort the <see cref="FeedPackageDetails"/> on.</param>
+        private async Task<SortedList<DateTime, IList<FeedPackageDetails>>> GetPackagesInOrder(
+            DateTime since,
+            int top,
+            Func<FeedPackageDetails, DateTime> keyDateFunc,
+            string columnName)
+        {
+            var allPackages = await GetPackages(since, top, columnName);
+
+            return OrderPackagesByKeyDate(allPackages, keyDateFunc);
+        }
+
+
+        /// <summary>
+        /// Returns a <see cref="SortedList{DateTime, IList{FeedPackageDetails}}"/> of packages.
+        /// </summary>
+        /// <param name="keyDateFunc">The <see cref="DateTime"/> field to sort the <see cref="FeedPackageDetails"/> on.</param>
+        public static SortedList<DateTime, IList<FeedPackageDetails>> OrderPackagesByKeyDate(
+            IReadOnlyCollection<FeedPackageDetails> packages,
+            Func<FeedPackageDetails, DateTime> keyDateFunc)
+        {
+            var result = new SortedList<DateTime, IList<FeedPackageDetails>>();
+
+            foreach (var package in packages)
+            {
+                var packageKeyDate = keyDateFunc(package);
+                if (!result.TryGetValue(packageKeyDate, out IList<FeedPackageDetails> packagesWithSameKeyDate))
+                {
+                    packagesWithSameKeyDate = new List<FeedPackageDetails>();
+                    result.Add(packageKeyDate, packagesWithSameKeyDate);
+                }
+
+                packagesWithSameKeyDate.Add(package);
+            }
+
+            return result;
+        }
+
+        /// <summary>
+        /// Asynchronously gets a <see cref="IList{FeedPackageDetails}"/> from the gallery database.
+        /// </summary>
+        /// <param name="connectionFactory">Provides a connection to the gallery database.</param>
+        /// <returns>A task that represents the asynchronous operation.
+        /// The task result (<see cref="Task{TResult}.Result" />) returns an
+        /// <see cref="IList{FeedPackageDetails}" />.</returns>
+        private async Task<IReadOnlyCollection<FeedPackageDetails>> GetPackages(
+            DateTime since,
+            int top,
+            string columnName)
+        {
+            using (var sqlConnection = await _connectionFactory.OpenAsync())
+            using (var sqlTransaction = sqlConnection.BeginTransaction())
+            {
+                var timestamps = await _db2catalogQueries.GetTimestampsAsync(
+                    sqlConnection,
+                    sqlTransaction,
+                    top,
+                    since,
+                    columnName);
+
+                return await _db2catalogQueries.GetPackageDetailsForTimestampsAsync(
```

Comment:
Would it be possible to just do the old style query for now and later introduce the `GROUP BY` improvement? Maybe we can phase it in to reduce risk.